src/CommandRunner.cpp

```diff
@@ -311,35 +311,41 @@ CommandRunner::CommandRunner(bool debug,
  *
  * @return The output of the command read from the output file.
  */
-Try<std::string> CommandRunner::run(const Command& command,
-                                    const std::string& input) {
+Try<string> CommandRunner::run(const Command& command,
+                               const std::string& input) {
   try {
-    TemporaryFile inputFile;
-    TemporaryFile outputFile;
-    inputFile.write(input);
+    TemporaryFile stdinFile;
+    TemporaryFile stdoutFile;
+    TemporaryFile stderrFile;
+    stdinFile.write(input);
 
     if (m_debug) {
       TASK_LOG(INFO, m_loggingMetadata)
           << "Calling command: \"" << command.command() << "\" ("
-          << command.timeout() << "s) " << inputFile.filepath() << " "
-          << outputFile.filepath();
+          << command.timeout() << "s) " << stdinFile.filepath() << " "
+          << stdoutFile.filepath() << " " << stderrFile.filepath();
     } else {
       TASK_LOG(INFO, m_loggingMetadata)
           << "Calling command: \"" << command.command() << "\" ("
           << command.timeout() << "s)";
     }
 
     vector<string> args;
-    args.push_back(inputFile.filepath());
-    args.push_back(outputFile.filepath());
+    args.push_back(stdinFile.filepath());
+    args.push_back(stdoutFile.filepath());
+    args.push_back(stderrFile.filepath());
 
     Try<Nothing> result = runCommandWithTimeout(
         command.command(), args, command.timeout(), m_loggingMetadata);
     if (result.isError()) {
+      string stderr = stderrFile.readAll();
```

Comment:
are we sure this file exists? and is readable? (and its output fits in memory :trollface: )

Reply:
No we are not, I will indeed make sure at least Mesos does not crash in such cases. I'll see what I can do for size in memory...

Reply:
ok